src/Illuminate/Database/Schema/Grammars/MySqlGrammar.php

```diff
@@ -492,12 +492,19 @@ public function compileSpatialIndex(Blueprint $blueprint, Fluent $command)
      */
     protected function compileKey(Blueprint $blueprint, Fluent $command, $type)
     {
-        return sprintf('alter table %s add %s %s%s(%s)',
+        $columns = collect($command->columns)->map(function ($column) {
+            // Check if the column contains a functional expression
+            // and return as-is, otherwise wrap
+            return preg_match('/\(.+\)/', $column) ? $column : $this->wrap($column);
+        })->implode(', ');
+
+        return sprintf(
+            'alter table %s add %s %s%s(%s)',
             $this->wrapTable($blueprint),
             $type,
             $this->wrap($command->index),
-            $command->algorithm ? ' using '.$command->algorithm : '',
-            $this->columnize($command->columns)
+            $command->algorithm ? ' using ' . $command->algorithm : '',
+            $columns
         );
     }
 
```

Comment:
One possible solution for https://github.com/laravel/framework/pull/54025#discussion_r1898460480
```suggestion
protected static function isFunctionalExpression(string $column): bool
{
    return preg_match('/\(.+\)/', $column);
}
```